doxia-integration-tools/src/main/java/org/apache/maven/doxia/tools/DefaultSiteTool.java

```diff
@@ -806,129 +790,98 @@ protected static String getNormalizedPath(String path) {
+            localeStr = language + "_" + country + "_" + variant;
+            ArtifactRequest request =
+                    createSiteDescriptorArtifactRequest(project, localeStr, remoteArtifactRepositories);
 
             try {
-                artifactResolver.resolve(artifact, repositories, localRepository);
-
-                siteDescriptor = artifact.getFile();
+                ArtifactResult result = repositorySystem.resolveArtifact(repoSession, request);
 
-                // we use zero length files to avoid re-resolution (see below)
-                if (siteDescriptor.length() > 0) {
-                    found = true;
-                } else {
+                // TODO Can "result.isMissing()" happen here?
+                siteDescriptor = result.getArtifact().getFile();
+                found = true;
+            } catch (ArtifactResolutionException e) {
+                if (e.getCause() instanceof ArtifactNotFoundException) {
                     LOGGER.debug("No site descriptor found for '" + project.getId() + "' for locale '" + localeStr
                             + "', trying without variant...");
+                } else {
+                    throw e;
                 }
-            } catch (ArtifactNotFoundException e) {
-                LOGGER.debug("Unable to locate site descriptor for locale '" + localeStr + "'", e);
-
-                // we can afford to write an empty descriptor here as we don't expect it to turn up later in the
-                // remote repository, because the parent was already released (and snapshots are updated
-                // automatically if changed)
-                siteDescriptor = new File(localRepository.getBasedir(), localRepository.pathOf(artifact));
-                siteDescriptor.getParentFile().mkdirs();
-                siteDescriptor.createNewFile();
             }
         }
 
         if (!found && !country.isEmpty()) {
-            String localeStr = language + "_" + country;
-            // TODO: this is a bit crude - proper type, or proper handling as metadata rather than an artifact in 2.1?
-            artifact = artifactFactory.createArtifactWithClassifier(
-                    project.getGroupId(), project.getArtifactId(), project.getVersion(), "xml", "site_" + localeStr);
+            localeStr = language + "_" + country;
+            ArtifactRequest request =
+                    createSiteDescriptorArtifactRequest(project, localeStr, remoteArtifactRepositories);
 
             try {
-                artifactResolver.resolve(artifact, repositories, localRepository);
+                ArtifactResult result = repositorySystem.resolveArtifact(repoSession, request);
 
-                siteDescriptor = artifact.getFile();
-
-                // we use zero length files to avoid re-resolution (see below)
-                if (siteDescriptor.length() > 0) {
-                    found = true;
-                } else {
+                // TODO Can "result.isMissing()" happen here?
```

Comment:
no - when not found exception will be thrown

Reply:
That is true, but if you look into Resolver code, there are spots which lead to this case there no exception is thrown, but in Maven Core code I wasn't able to find any `result.isMissing()` for a non-exceptional case. Maybe, I can ignore it.

Reply:
IMHO always throw

https://github.com/apache/maven-resolver/blob/master/maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultArtifactResolver.java#L440-L457

Reply:
But @cstamas is expert here 😄

Reply:
What about https://github.com/apache/maven-resolver/blob/cb2923f31b940eab699743e8398426b09a6cfc62/maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultArtifactResolver.java#L403-L413?

Reply:
hym ... in 403 artifact.file will be not set so checks in 444 should throw ... but who know

Reply:
https://github.com/apache/maven-resolver/blob/cb2923f31b940eab699743e8398426b09a6cfc62/maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultArtifactResolver.java#L456 yup

Reply:
WITHOUT exception

Reply:
You describe the case WHEN an exception is thrown. I talk about the opposite.

Reply:
I tried the offline case, no file means failure with exception. I will remove the TODOs until proven otherwise.